tools/MakeNiiFilesBIDSCompliant.pl

```diff
@@ -136,637 +138,1832 @@ =head2 Methods
-    per site basis.
-    - Need to add to the multi-echo sequences a JSON file with the echo time within,
-    as well as the originator NIfTI parent file. In addition, we need to check from
-    the database if the sequence is indeed a multi-echo and require the
-    C<BIDSMultiEcho> column set by the project in the C<bids_mri_scan_type_rel>
-    table.
+Adds an entry in the participants.tsv BIDS file for a given candidate.
 
-=head1 COPYRIGHT AND LICENSE
+INPUTS:
+    - $minc_file_hash: hash with information associated to the MINC file
+    - $bids_root_dir : path to the BIDS root directory
+    - $db_handle     : database handle
 
-License: GPLv3
+=cut
 
-=head1 AUTHORS
+sub add_entry_in_participants_bids_file {
+    my ($minc_file_hash, $bids_root_dir, $db_handle) = @_;
+
+    my $participants_tsv_file  = $bids_root_dir . '/participants.tsv';
+    my $participants_json_file = $bids_root_dir . '/participants.json';
+
+    my $cand_id = $minc_file_hash->{'candID'};
+
+    if (! -e $participants_tsv_file) {
+        # create the tsv and json file if they do not exist
+        create_participants_tsv_and_json_file($participants_tsv_file, $participants_json_file);
+        registerBidsFileInDatabase(
+            $participants_tsv_file,  'study',                  'tsv',  undef,
+            undef,                   'participants_list_file', undef
+        );
+        registerBidsFileInDatabase(
+            $participants_json_file, 'study',                  'json', undef,
+            undef,                   'participants_list_file', undef
+        );
+    } else {
+        # read participants.tsv file and check if a row is already present for
+        # that subject
+        open (FH, '<:encoding(utf8)', $participants_tsv_file) or die " $!";
+        while (my $row = <FH>) {
+            return if ($row =~ m/^sub-$cand_id/);
+        }
+    }
 
-LORIS community <loris.info@mcin.ca> and McGill Centre for Integrative Neuroscience
+    # grep the values to insert in the participants.tsv file
+    my $values = grep_participants_values_from_db($db_handle, $cand_id);
+    open (FH, '>>:encoding(utf8)', $participants_tsv_file) or die " $!";
+    print FH (join("\t", @$_), "\n") for $values;
+    close FH;
+}
+
+
+=pod
+
+=head3 grep_participants_values_from_db($db_handle, $cand_id)
+
+Gets participant's sex from the candidate table.
+
+INPUTS:
+    - $db_handle: database handle
+    - $cand_id  : candidate ID
+
+OUTPUT:
+    - @values: array with values returned from the candidate table
 
 =cut
+
+sub grep_participants_values_from_db {
+    my ($db_handle, $cand_id) = @_;
+
+    my $candidate_query = "SELECT Sex FROM candidate WHERE CandID = ?";
+
+    my $st_handle = $db_handle->prepare($candidate_query);
+    $st_handle->execute($cand_id);
+
+    my @values = $st_handle->fetchrow_array;
+    unshift(@values, "sub-$cand_id");
+
+    # TODO: could add registration site , project to the file
+    return \@values;
```

Comment:
in participants.tsv, is there a value to add registration site, registration project? Other fields from the candidate table?